zend_abstract_interface/symbols/lookup.c

```diff
@@ -77,42 +77,20 @@ static inline zai_str zai_symbol_lookup_clean(zai_str view) {
     return ZAI_STR_NEW(view.ptr + 1, view.len - 1);
 }
 
-static inline zai_str zai_symbol_lookup_key(zai_str *namespace, zai_str *name, bool lower) {
-    zai_str rv;
-    zai_str vns     = zai_symbol_lookup_clean(*namespace);
-    zai_str vn      = zai_symbol_lookup_clean(*name);
-
-    char *result = NULL;
-#define CHAR_AT(n) (result[n])
-    if (vns.len) {
-        rv.len = vns.len + vn.len + 1;
-        result = pemalloc(rv.len + 1, 1);
-
-        memcpy(&CHAR_AT(0), vns.ptr, vns.len);
-
-        for (uint32_t c = 0; c < vns.len; c++) {
-            CHAR_AT(c) = tolower(CHAR_AT(c));
-        }
-
-        CHAR_AT(vns.len) = '\\';
-        memcpy(&CHAR_AT(vns.len + 1), vn.ptr, vn.len);
-    } else {
-        rv.len = vn.len;
-        result = pemalloc(rv.len + 1, 1);
-
-        memcpy(&CHAR_AT(0), vn.ptr, vn.len);
+static zai_string zai_symbol_lookup_key(zai_str *namespace, zai_str *name, bool lower) {
+    zai_str vns         = zai_symbol_lookup_clean(*namespace);
+    zai_str separator   = vns.len ? ZAI_STRL("\\") : ZAI_STR_EMPTY;
+    zai_str vn          = zai_symbol_lookup_clean(*name);
+    zai_string rv       = zai_string_concat3(vns, separator, vn);
+
+    // Namespaces are never case-sensitive, so they are always lowered, even
```

Comment:
You seem to be re-lowering them all the same.

Reply:
Yes, we always have to lower the name because they are always case insensitive. The user may lookup `"MyFoo\Bar"`, or `myfoo\Bar`, and it's the same.